bin/create-map-file

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import os
+import click
+import click_log
+import logging
+
+import ROOT
+from rootpy.io import root_open
+from rootpy.tree import Tree
+from cmsl1t.utils.module import load_L1TNTupleLibrary
+import cmsl1t
+
+import yaml
+import collections
+
+logger = logging.getLogger(__name__)
+logging.getLogger("rootpy.tree.chain").setLevel(logging.WARNING)
+click_log.basic_config(logger)
+
+ALIAS_REGISTRY = []
+
+
+def extract_branches(tree):
+    paths = []
+    for branch in tree.branches:
+        branchName = branch.GetName()
+
+        if isinstance(branch, ROOT.TBranchElement):
+            leaves = ['.'.join([branchName, l.GetName()])
+                      for l in branch.GetListOfBranches()]
+            if leaves:
+                paths += leaves
+            else:
+                paths.append(branchName)
+
+    return sorted(paths, key=lambda s: s.lower())
+
+
+def extract_trees(input_file):
+    trees = {}
+
+    with root_open(input_file) as f:
+        for path, dirs, objects in f.walk():
+            for objName in objects:
+                objPath = os.path.join(path, objName)
+                obj = f.Get(objPath)
+                if isinstance(obj, Tree):
+                    trees[objPath] = dict(
+                        name=objName,
+                        branches=extract_branches(obj),
+                    )
+        return trees
+
+
+def convert_to_dict(trees):
+    trees_dict = {}
+    for path, tree in trees.items():
+        trees_dict[path] = dict(
+            name=tree['name'],
+            branches={b: {} for b in tree['branches']},
+        )
+    return trees_dict
+
+
+def _full_path_alias(path, objName):
+    # path includes treeName
+    tokens = path.split('/')
+    tokens += objName.split('.')
+    return 'event.' + '_'.join(tokens)
+
+
+def _default_alias(path, treeName, objName):
+    tokens = []
+    if 'emu' in path.lower():
+        tokens.append('emu')
+    tokens.append(treeName)
+    tokens += objName.split('.')
+    return 'event.' + '_'.join(tokens)
+
+
+def _shorthand_alias(path, treeName, objName):
+    tokens = []
+    path_lower = path.lower()
+    obj_lower = objName.lower()
+    search_for = ['emu', 'bmt', 'emt', 'omt']
+    for s in search_for:
+        if s in path_lower or s in obj_lower:
+            tokens.append(s)
+    tokens += objName.split('.')[-2:]
+    return 'event.' + '_'.join(tokens)
+
+
+def getDefaultAliases(path, treeName, objName):
+    alias1 = _full_path_alias(path, objName)
+    alias2 = _default_alias(path, treeName, objName)
+    alias3 = _shorthand_alias(path, treeName, objName)
```

Comment:
I feel like having three aliases for each branch is overkill and might lead to confusion.  I'd push for having a single "alias" and also provide access to the full alias, although have this slightly more hidden in the yaml file?